src/adapters/scene.js

```diff
@@ -143,6 +147,92 @@ Scene.prototype.initMapBox = function () {
   })
 }
 
+Scene.prototype.isPointInBounds = function(point, bounds) {
+  var lng = (point.lng - bounds._ne.lng) * (point.lng - bounds._sw.lng) < 0;
+  var lat = (point.lat - bounds._ne.lat) * (point.lat - bounds._sw.lat) < 0;
+  return lng && lat;
+}
+
+Scene.prototype.isBBoxInExtendedViewport = function(bbox){
+
+  // Get viewport bounds
+  var viewport = this.mb.getBounds();
+
+  // Compute "width", "height"
+  var width = viewport._ne.lng - viewport._sw.lng;
+  var height = viewport._ne.lat - viewport._sw.lat;
+
+  // Compute extended viewport
+  viewport._ne.lng += width;
+  viewport._ne.lat += height;
+
+  viewport._sw.lng -= width;
+  viewport._sw.lat -= height;
+
+  // Check bounds:
+
+  // Lng between -180 and 180 (wraps: 180 + 1 = -179)
+  if(viewport._ne.lng < 180) viewport._ne.lng += 360;
+  if(viewport._ne.lng > 180) viewport._ne.lng -= 360;
+
+  if(viewport._sw.lng < 180) viewport._sw.lng += 360;
+  if(viewport._sw.lng > 180) viewport._sw.lng -= 360;
+
+  // Lat between -85 and 85 (does not wrap)
+  if(viewport._ne.lat < -85) viewport._ne.lat = -85;
+  if(viewport._ne.lat > 85) viewport._ne.lat = 85;
+
+  if(viewport._sw.lat < -85) viewport._sw.lat = -85;
+  if(viewport._sw.lat > 85) viewport._sw.lat = 85;
+
+
+  // Check if one corner of the BBox is in the extended viewport:
+
+  var bboxIsNear = false;
+
+  // ne
+  if(this.isPointInBounds(bbox._ne, viewport)){
+    bboxIsNear = true;
+  }
+
+  // nw
+  if(this.isPointInBounds({lng: bbox._sw.lng, lat: bbox._ne.lat}, viewport)){
+    bboxIsNear = true;
+  }
+
+  // se
+  if(this.isPointInBounds({lng: bbox._ne.lng, lat: bbox._sw.lat}, viewport)){
+    bboxIsNear = true;
+  }
+
+  // sw
+  if(this.isPointInBounds(bbox._sw, viewport)){
+    bboxIsNear = true;
+  }
+
+  return bboxIsNear
+
+}
+
+Scene.prototype.fitBbox = function(bbox, padding){
+
+  padding = padding || {left: 0, top: 0, right: 0, bottom: 0};
+
+  console.log(padding);
+
+  // Animate if the zoom is big enough and if the BBox is (partially or fully) in the extended viewport
+  var animate = false;
+
+  if(this.mb.getZoom() > 10 && this.isBBoxInExtendedViewport(bbox)){
```

Comment:
this logic already exist

Reply:
yep, it's here until we merge the functions